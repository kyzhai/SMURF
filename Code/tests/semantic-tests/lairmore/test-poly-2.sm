f :: a -> b

f x = []

y = f 1

main = []

