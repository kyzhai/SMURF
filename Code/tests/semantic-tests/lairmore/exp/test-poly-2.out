Program: 
	SFuncdec: 
		f (x) :: a -> b = []
	New Scope: 
		ID: x  :: a

	
	SVardef: 
		ID: y  :: []

	f 1
	SMain: []
Global Scope: 
	ID: main  :: []
	ID: f x :: a -> b
	ID: y  :: []
	ID: print x :: a -> a
	ID: random  :: Int
	ID: head h : t :: [a] -> a
	ID: tail h : t :: [a] -> [a]

